engine/storage/snapshot/src/org/apache/cloudstack/storage/snapshot/SnapshotObject.java

```diff
@@ -149,6 +149,17 @@ public long getPhysicalSize() {
         return physicalSize;
     }
 
+    @Override
+    public void markBackedUp() throws CloudRuntimeException{
+        try {
+            processEvent(Event.OperationNotPerformed);
+        } catch (NoTransitionException ex) {
+            s_logger.error("no transition error: ", ex);
+            throw new CloudRuntimeException("Error marking snapshot backed up: " +
```

Comment:
Can you maintain the history of the stack of the exception that is re-thrown here?
I mean, it is a matter of using `throw new CloudRuntimeException(<message>, <exception>)`

Reply:
@nathanejohnson I see this point as critical. Doing as @rafaelweingartner proposed would allow having the full stack on the log, which helps on debugging.
Could you please address this point? Thanks! Overall LGTM.